### PR TITLE
Fix decoding of JPX images having an alpha channel

### LINKED ISSUE #11306: PDF Annotations made in iOS do not render properly
Attach (recommended) or Link to PDF file here: 
[2018 Wrapped.pdf](https://github.com/mozilla/pdf.js/files/3806127/2018.Wrapped.pdf)

Configuration:
- Web browser and its version: Chrome 78.0.3904.70 (Official Build) (64-bit)
- Operating system and its version: MacOS 10.14.6 
- PDF.js version: v2.2.228 and the latest version in PDF.js demo (https://mozilla.github.io/pdf.js/web/viewer.html)
- Is a browser extension: No

Steps to reproduce the problem:
1. Annotate PDF on iOS
2. Open it with PDF.js (https://mozilla.github.io/pdf.js/web/viewer.html)

What is the expected behavior? (add screenshot)
image

What went wrong? (add screenshot)
image (1)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### LINKED ISSUE #16782: JPX Image with transparency shows inversed colors
with the following PDF:
[tt1.pdf](https://github.com/mozilla/pdf.js/files/12244643/tt1.pdf)

we are expecting this image:(available in AcrobatReader, Chrome, Sumatra,...)
image

but in pdf.js (windows 11 / firefox 115.02 + PDF.js: 3.7.96 [23958ffc5] the colors are inverted:

### LINKED ISSUE #18194: Wrong rendering of JPX images with transparency
Attach (recommended) or Link to PDF file here:
[out2.pdf](https://github.com/user-attachments/files/15511847/out2.pdf)

File is from https://gitlab.freedesktop.org/poppler/poppler/-/issues/1486

Configuration:
- Web browser and its version:
FF 126.0.1
- Operating system and its version:
Windows  10.0.19045.4474
- PDF.js version:
latest https://mozilla.github.io/pdf.js/web/viewer.html

Steps to reproduce the problem:
1. open the file

What is the expected behavior? (add screenshot)
grafik

What went wrong? (add screenshot)
grafik

### PR DESCRIPTION
**Pull request by calixteman**
When an image has a non-zero SMaskInData it means that the image has an alpha channel.
With JPX images, the colorspace isn't required (by spec) so when we don't have it, the JPX decoder will handle the conversion in RGBA format.

### CODE REVIEW

**Review by Snuffleupagus — APPROVED** [review]
r=me, with a couple of nits; thank you!

**Comment by timvandermeij** [comment]
It looks like there is a regression in `(chrome|firefox)-issue11004-page1` and `(chrome|firefox)issue12579-page3` since they don't seem to render at all anymore in the reference tests. (With that fixed though I'm also +1 on this patch.)

File: src/core/decode_stream.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ class DecodeStream extends BaseStream {
     return this.buffer[this.pos++];
   }
 
-  getBytes(length, ignoreColorSpace = false) {
+  getBytes(length, decoderOptions = null) {
```
Comment (timvandermeij) [inline src/core/decode_stream.js:76]: It looks like setting this to `null` in the latest patch is causing the two mentioned regressions. Checking out this PR and opening that document shows this in the console: `Warning: Unable to decode image "img_p3_1": "TypeError: (destructured parameter) is null"`.

**Pull request merged**